### PR TITLE
Stacking notifications with different types

### PR DESCRIPTION
**Pull request by andygnewman**
Objective:
Handle multiple notifications concurrently (with an animation that suits concurrent notifications)
Enable accessibility
Differentiate styling by type (eg error, success as well as default)
Simplify the module;
- remove ability to call hide directly (only timeout / click close button)
- remove ability to not show close button

nnotification

A notification of type = success
screen shot 2016-01-28 at 17 44 05

(To be released as a major version)

To Do;
- [x] Handle multiple notices stacked in order of receipt (most recent at top)
- [x] Add role=alert so screen readers will announce notification
- [x] Have default styling (green) and error styling (red)
- [x] Update tests
- [x] Update the README

@phamann

### CODE REVIEW

**Comment by phamann** [comment]
As discussed. Would good if we could keep:
- The instantiation via custom events to encourage de-coupling of the component from others.
- The animation, but make it optional via a configuration parameter
- a notification `type` such as `error`, `success` and `info`

File: main.js [hunk 1]
```diff
@@ -1,78 +1,61 @@
...
-	if(!called){
-		func();
-		called = true;
-	}
-}
+		options.trackable = options.trackable || `notification-${options.type}`;
 
-function show(options){
-	clearTimeout(timeout);
+		this.notice = document.createElement('div');
```
Comment (phamann) [inline main.js:15]: Is a `div` necessary here? May be good use of a [`new DocumentFragment`](https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment) instead.

Comment (andygnewman) [inline main.js:15]: Nice suggestion. Had a go with it, but once you load the fragments elements into the DOM you don't have any reference to the fragment, so you can't easily get rid of the notice when it's time is up.

Comment (phamann) [inline main.js:15]: Ah yes, makes sense :+1:

**Comment by phamann** [comment]
:+1: Nice

**Pull request merged**